src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/PseudoCustomAttributesData.cs

```diff
@@ -186,9 +183,9 @@ internal BlobBuilder SerializeMarshallingData()
                     {
                         writer.WriteCompressedInteger((int)safeArrayElementSubtype);
 
-                        if (_marshalTypeNameOrSymbol is Type elementType)
+                        if (_marshalTypeName is string)
```

Comment:
```suggestion
                        if (_marshalTypeName != null)
```
Nit: Match the style used in the rest of the method.